doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+# StatusPort
+
+## Summary and requirements
+
+The `StatusPort` is an alternative to the publish subscriber communication in `iceoryx_posh`.
+
+The target of the `StatusPort` are use-case with the following properties:
+
+* Data is transferred once or is rarely updated
+* There are many readers which are interested in the data
+* Data can be persistent
+    * The lifetime of the transferred data is bound to the lifetime of `StatusPortData`
+* Readers don't access the data directly but via a lambda
+    * Preventing torn reads, since the `StatusPortReader` detects if the data
+    changed during `take()` operation (Frankenstein check) and re-executes the lambda
+* Has to be attachable to `Listener`
+* `StatusPort` could potentially be used to communicate between different ASIL domains
+
+Potential applications are:
+
+* Introspection topics
+* Service discovery
+
+## Design
+
+### Terminology
+
+* Transaction: Atomic state of the world, which is changed by a write operation
+* Current transaction: Transaction in local scope, read in the beginning of each operation
+* Latest transaction: Transaction in the shared memory managment segment
+* Chunk: Untyped piece of memory located in the shared memory payload segment
+* Read position: The chunk, which was used the last time to write data
+* Write position: The opposite chunk, not currently being used by the `StatusPortReader`s
+
+### Contract & properties
+
+* 1:n communication aka broadcast
+* Two memory chunks from shared memory payload segment are used
+* Data exchanged needs to be trivially-copyable
+* Storing a chunk cannot fail
+* Reading a chunk cannot fail
+    * If no data was sent yet, `callable` is not called
+    * If data was updated in-between read, re-call `callable`
+* Not part of the user API in the first version, only used internally
+    * Might become user API for freedom-from-interference separation
+* Reader only needs write access to the data structure and no read access
+* Writing and Reading will be tried indefinitely till possible, hence starvation
+is possible
+* `StatusPortData` is created in the shared memory segment if either a `StatusPortWriter`
+or `StatusPortReader` is created
+    * Two chunks in the shared memory payload segment are bound to the lifetime
+    of the `StatusPortData` (either via `StatusPortData` c'tor or
+    `StatusPort{Writer,Reader}` c'tor)
+
+### Discarded ideas & design alternatives
```

Comment:
I agree, otherwise this design documented is a little bit bloated.

Reply:
Ok, got your point and agree. I put these alternatives here to prevent questions "Why didn't you do x?". If you don't mind I'd keep this section as it is and make sure with the next design document to keep the alternative section short and precise.